File: typescript-examples/stagehand-template/hooks/setupContext.ts
import { Stagehand } from "@browserbasehq/stagehand";
import { attemptStore } from "@intuned/runtime";

async function getWebSocketUrl(cdpUrl: string) {
  const versionUrl = cdpUrl.endsWith("/")
    ? `${cdpUrl}json/version`
    : `${cdpUrl}/json/version`;
  const response = await fetch(versionUrl);
  const data = await response.json();
  return data.webSocketDebuggerUrl;
}

export default async function setupContext({
  cdpUrl,
}: {
  cdpUrl: string;
  apiName: string;
  apiParameters: any;
}) {
  const webSocketUrl = await getWebSocketUrl(cdpUrl);
  const stagehand = new Stagehand({
    env: "LOCAL",
    localBrowserLaunchOptions: {
      cdpUrl: webSocketUrl,
    },

    // this is needed since Stagehand uses pino which spawns a worker thread, which is not currently supported in Intuned
    logger: console.log,
  });

  await stagehand.init();

  attemptStore.set("stagehand", stagehand);

  return {
    context: stagehand.context as any,
    page: stagehand.context.pages()[0] as any,
    cleanup: async () => {
      await stagehand.close();
    },
  };
}

File: python-examples/computer-use/hooks/setup_context.py
from intuned_runtime import attempt_store
from stagehand import Stagehand


async def setup_context(*, api_name: str, api_parameters: str, cdp_url: str):
    if api_name == "api/gemini-computer-use":
        stagehand = Stagehand(
            env="LOCAL", local_browser_launch_options=dict(cdp_url=cdp_url)
        )
        await stagehand.init()
        attempt_store.set("stagehand", stagehand)

        async def cleanup():
            await stagehand.close()

        return stagehand.context, stagehand.page, cleanup

    elif api_name == "api/browser-use":
        from browser_use import Browser

        browser = Browser(cdp_url=cdp_url)
        attempt_store.set("browser", browser)
        return

        
        
